more DQM-related developments, fixes and improvements

fwyzard opened the pull request:
Follow up of the work in #21652 .

Rename private `book...` methods to `book..._` to reduce the ambiguity of the public vs private interface.

Update the `ConcurrentMonitorElement` class:
  - fix the `lock_guard` used in the move constructor
  - use a `tbb:spin_mutex` in place of a `std::mutex`

On a microbenchmark, when guarding a small code section `tbb:spin_mutex` is twice as fast as `std::mutex` if there is no contention, and even faster as the number of threads increase.

fwyzard committed:
DQMStore: rename private `book...` methods

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-21729/2618

cmsbuild commented:
A new Pull Request was created by  @fwyzard (Andrea Bocci) for master.

It involves the following packages:

DQMServices/Components
DQMServices/Core
DQMServices/FileIO


@vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien can you please review it and eventually sign? Thanks.
@barvic this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 1d6f718659d4183273e09c259b13965694d7f3d8
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25125/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25125/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins-workarea/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21729/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 15 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2836825
 - DQMHistoTests: Total failures: 1566
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2835081
 - DQMHistoTests: Total skipped: 178
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.35000000009 KiB( 23 files compared)
 - Checked 113 log files, 9 edm output root files, 27 DQM output files

fwyzard committed:
MonitorElement: cleanup

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-21729/2631

cmsbuild commented:
Pull request #21729 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d830b59b08a446c6f65fc3bb3d0809f2b2db7292
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25145/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25145/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins-workarea/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21729/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 16 differences found in the comparisons
 - DQMHistoTests: Total files compared: 27
 - DQMHistoTests: Total histograms compared: 2836825
 - DQMHistoTests: Total failures: 2119
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2834528
 - DQMHistoTests: Total skipped: 178
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.02000000015 KiB( 23 files compared)
 - Checked 113 log files, 9 edm output root files, 27 DQM output files

fwyzard commented:
hold

cmsbuild commented:
Pull request has been put on hold by  @fwyzard
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

fwyzard committed:
ConcurrentMonitorElement: fix lock in move constructor

fwyzard committed:
ConcurrentMonitorElement: use tbb:spin_mutex in place of std::mutex

On a microbenchmark, `tbb:spin_mutex` is twice as fast as `std::mutex`
when there is no contention, and even faster if there is contendtion,
when guarding a small code section.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-21729/2798

cmsbuild commented:
Pull request #21729 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-21729/2799

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 616d575f1861c9b6c8d1708ce5149ff215c86763
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25321/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-21729/25321/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins-workarea/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-21729/1325.7_TTbar_13_94XNanoAODINPUT+TTbar_13_94XNanoAODINPUT+NANOEDMMC2017

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2775243
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2775073
 - DQMHistoTests: Total skipped: 169
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.08999999974 KiB( 22 files compared)
 - Checked 110 log files, 9 edm output root files, 26 DQM output files

fwyzard commented:
@vazzolini, @kmaeshima, @dmitrijus, could you review and sign this before the release planning meeting tomorrow ?

dmitrijus commented:
+1

Thanks a lot, this is very welcome change :)

fwyzard commented:
unhold

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)
